Tasks/SonarQubePostTest/PRCA/PostComments-Module.psm1

```diff
@@ -334,8 +348,8 @@ function GetMatchingComments
      $matchingThreads = $existingThreads | Where-Object {
             ($_ -ne $null) -and
             ($_.Status -ne [Microsoft.VisualStudio.Services.CodeReview.Discussion.WebApi.DiscussionStatus]::Fixed) -and
-            (ThreadMatchesCommentSource $_ $script:messageSource) -and
-            (ThreadMatchesItemPath $_ $message.RelativePath)}
+            ($_.ItemPath -eq $message.RelativePath) -and
```

Comment:
The DiscussionThread object has a read-only ItemPath string property. It is probably set when the server persists the thread object. It should be used instead of the thread.Properties[ItemPath] since it works for both legacy and code flow operations.